packages/wxt/src/core/builders/vite/plugins/devHtmlPrerender.ts

```diff
@@ -113,29 +113,35 @@ export function devHtmlPrerender(
     {
       name: 'wxt:virtualize-inline-scripts',
       apply: 'serve',
-      resolveId(id) {
-        // Resolve inline scripts
-        if (id.startsWith(virtualInlineScript)) {
-          return '\0' + id;
-        }
+      resolveId: {
+        filter: {
+          id: [new RegExp(`^${virtualInlineScript}`), new RegExp('^/chunks/')],
+        },
+        handler(id) {
+          // Ignore chunks during HTML file pre-rendering
+          if (id.startsWith('/chunks/')) {
+            return '\0noop';
+          }
 
-        // Ignore chunks during HTML file pre-rendering
-        if (id.startsWith('/chunks/')) {
-          return '\0noop';
-        }
+          return '\0' + id;
+        },
       },
-      load(id) {
-        // Resolve virtualized inline scripts
-        if (id.startsWith(resolvedVirtualInlineScript)) {
-          // id="virtual:wxt-inline-script?<key>"
+      load: {
+        filter: {
+          id: [
+            new RegExp(`^${resolvedVirtualInlineScript}`),
+            new RegExp('^\x00noop'),
+          ],
+        },
+        handler(id) {
+          // Ignore chunks during HTML file pre-rendering
+          if (id === '\0noop') {
+            return '';
+          }
+
           const key = id.substring(id.indexOf('?') + 1);
```

Comment:
Add this comment back

```suggestion
          // id="virtual:wxt-inline-script?<key>"
          const key = id.substring(id.indexOf('?') + 1);
```